### PR TITLE
update chargeitem create

### PR DESCRIPTION
**Pull request by krvicky31**
Description
----

PR Checklist
----
- [x] Screenshot(s) of changes attached before changes merged.
Screenshot 2021-12-14 at 3 49 51 PM
Screenshot 2021-12-14 at 3 50 11 PM
Screenshot 2021-12-14 at 3 50 38 PM
Screenshot 2021-12-14 at 3 50 56 PM
Screenshot 2021-12-14 at 3 51 14 PM
Screenshot 2021-12-14 at 3 51 26 PM
Screenshot 2021-12-14 at 3 51 54 PM
Screenshot 2021-12-14 at 3 52 08 PM
Screenshot 2021-12-14 at 3 52 25 PM
Screenshot 2021-12-14 at 3 52 37 PM
Screenshot 2021-12-14 at 3 52 54 PM
Screenshot 2021-12-14 at 3 53 06 PM
Screenshot 2021-12-14 at 3 53 22 PM
Screenshot 2021-12-14 at 3 53 40 PM
Screenshot 2021-12-14 at 3 53 59 PM
Screenshot 2021-12-14 at 3 54 15 PM
Screenshot 2021-12-14 at 3 54 30 PM
Screenshot 2021-12-14 at 3 54 43 PM

- [ ] Screenshot(s) of changes attached after changes merged and published.

### CODE REVIEW

File: lib/resources/r4/charge_item_create.yaml [hunk 1]
```diff
@@ -0,0 +1,425 @@
...
+    type: resource
+    description: The ChargeItem to be created.
+    action: create
+    example: |
+      {
+        "parameter": [
+          {
+            "name": "chargeItem",
+            "resource": {
+              "resourceType": "chargeItem"
```
Comment (SantanuBanerjee) [inline lib/resources/r4/charge_item_create.yaml:36]: I think the resourceType should be ChargeItem with capital C

File: content/millennium/r4/financial/general/charge-item.md [hunk 2]
```diff
@@ -212,6 +212,61 @@ The `ETag` response header indicates the current `If-Match` version to use on su
...
+
+### Body Fields
+
+<%= definition_table(:charge_item_create, :create, :r4) %>
+
+### Example
+
+#### Request
+
+    POST https://fhir-ehr.cerner.com/r4/2c400054-42d8-4e74-87b7-80b5bd5fde9f/ChargeItem/$create
```
Comment (SantanuBanerjee) [inline content/millennium/r4/financial/general/charge-item.md:243]: The tenant key provided here should be from prod. Currently, it is from dev.

File: lib/resources/r4/charge_item_create.yaml [hunk 3]
```diff
@@ -0,0 +1,425 @@
...
+          {
+            "valueInteger": 68985657,
+            "url": "custom-attribute-value"
+          }
+        ],
+        "url": "https://fhir-ehr.cerner.com/r4/StructureDefinition/custom-attribute"
+      }
+    note: >
+      <ul>
+        <li>Custom-attribute-value and Custom-attribute-value are both required.</li>
```
Comment (alexbeckner1) [inline lib/resources/r4/charge_item_create.yaml:402]: I think one of these should be custom-attribute-name.

File: lib/resources/r4/charge_item_create.yaml (hunk 1 above)
Comment (alexbeckner1) [reply lib/resources/r4/charge_item_create.yaml:36]: I think we ended up going lowercase here to match the transactionId parameter. Including @exopsRamba to confirm.

File: content/millennium/r4/financial/general/charge-item.md (hunk 2 above)
Comment (alexbeckner1) [reply content/millennium/r4/financial/general/charge-item.md:243]: I believe ec2458f2-1e24-41c8-b71b-0e701af7583d is the C1941 tenant that should be used.

File: lib/resources/r4/charge_item_create.yaml [hunk 4]
```diff
@@ -0,0 +1,425 @@
...
+        <li>Custom-attribute-value can be of type integer, string, decimal, or date.</li>
```
Comment (alexbeckner1) [inline lib/resources/r4/charge_item_create.yaml:403]: date is going to end up changing to dateTime

File: lib/resources/r4/charge_item_create.yaml (hunk 1 above)
Comment (SantanuBanerjee) [reply lib/resources/r4/charge_item_create.yaml:36]: oh, I thought we wanted only the transaction name to be "chargeItem" with small C, which already is.

File: content/millennium/r4/financial/general/charge-item.md (hunk 2 above)
Comment (krvicky31) [reply content/millennium/r4/financial/general/charge-item.md]: Updated all where applicable  [Fixed](https://github.com/cerner/fhir.cerner.com/pull/778/files#diff-2830ed2523973e82037a23daf274840e05a6c6e05b9f698fbd5f3e7fc24f57bbR245)

File: lib/resources/r4/charge_item_create.yaml (hunk 3 above)
Comment (krvicky31) [reply lib/resources/r4/charge_item_create.yaml]: [Updated here](https://github.com/cerner/fhir.cerner.com/pull/778/files#diff-2830ed2523973e82037a23daf274840e05a6c6e05b9f698fbd5f3e7fc24f57bbR402)

File: lib/resources/r4/charge_item_create.yaml (hunk 4 above)
Comment (krvicky31) [reply lib/resources/r4/charge_item_create.yaml]: [Fixed here](https://github.com/cerner/fhir.cerner.com/pull/778/files#diff-2830ed2523973e82037a23daf274840e05a6c6e05b9f698fbd5f3e7fc24f57bbR403)

File: lib/resources/r4/charge_item_create.yaml [hunk 5]
```diff
@@ -0,0 +1,425 @@
...
+            },
+            "actor": {
+              "reference": "Practitioner/190182805"
+            }
+          }
+        ]
+      }
+    note: >
+      <ul>
+        <li>Need notes around supported system and values for function</li>
```
[the remark this answers is not in the document]
Comment (krvicky31) [reply lib/resources/r4/charge_item_create.yaml:179]: @BhavyaAcharya Can you look and gather the data which needs to be updated here.

File: lib/resources/r4/charge_item_create.yaml (hunk 1 above)
Comment (krvicky31) [reply lib/resources/r4/charge_item_create.yaml:36]: So, it will remain the same.

Comment (alexbeckner1) [reply lib/resources/r4/charge_item_create.yaml:36]: Sorry. I was wrong. Santanu is correct. name = chargeItem and resourceType = ChargeItem

File: lib/resources/example_json/r4_examples_charge_item.rb [hunk 6]
```diff
@@ -513,5 +513,221 @@ module Resources
...
+    R4_CHARGE_ITEM_CREATE ||= {
+      'resourceType': 'Parameters',
+      'parameter': [
+        {
+          'name': 'chargeItem',
+          'resource': {
+            'resourceType': 'ChargeItem',
+            'status': 'unknown',
+            'subject': {
+              'reference': 'Patient/178594204'
```
Comment (alexbeckner1) [inline lib/resources/example_json/r4_examples_charge_item.rb:526]: Let's use patient 12742399

File: lib/resources/example_json/r4_examples_charge_item.rb [hunk 7]
```diff
@@ -513,5 +513,221 @@ module Resources
...
+            },
+            'context': {
+              'reference': 'Encounter/53066085'
```
Comment (alexbeckner1) [inline lib/resources/example_json/r4_examples_charge_item.rb:529]: Let's use encounter 97953601

File: lib/resources/example_json/r4_examples_charge_item.rb [hunk 8]
```diff
@@ -513,5 +513,221 @@ module Resources
...
+              ]
+            },
+            'performer': [
+              {
+                'function': {
+                  'coding': [
+                    {
+                      'system': 'http://terminology.hl7.org/CodeSystem/v2-0912',
+                      'code': 'OP',
+                      'display': 'Ordering Provider'
```
Comment (alexbeckner1) [inline lib/resources/example_json/r4_examples_charge_item.rb:559]: Do we actually support/validate the display and the text for the performer function on a create?

File: lib/resources/example_json/r4_examples_charge_item.rb [hunk 9]
```diff
@@ -513,5 +513,221 @@ module Resources
...
+                    }
+                  ],
+                  'text': 'Ordering Provider'
+                },
+                'actor': {
+                  'reference': 'Practitioner/190182805'
```
Comment (alexbeckner1) [inline lib/resources/example_json/r4_examples_charge_item.rb:565]: Let's use practitioner 11817978

File: lib/resources/example_json/r4_examples_charge_item.rb [hunk 10]
```diff
@@ -513,5 +513,221 @@ module Resources
...
+                      {
+                        'valueCoding': {
+                          'system': 'https://fhir.cerner.com/ec2458f2-1e24-41c8-b71b-0e701af7583d/codeSet/14002',
+                          'code': '3693'
+                        },
+                        'url': 'https://fhir-ehr.cerner.com/r4/StructureDefinition/bill-code-schedule'
+                      }
+                    ],
+                    'system': 'https://fhir.cerner.com/ec2458f2-1e24-41c8-b71b-0e701af7583d/CodeSystem/BillCodes-ICD',
+                    'code': 'Y22'
```
Comment (alexbeckner1) [inline lib/resources/example_json/r4_examples_charge_item.rb:589]: Let's use code Z00 here

File: lib/resources/example_json/r4_examples_charge_item.rb [hunk 11]
```diff
@@ -513,5 +513,221 @@ module Resources
...
+                  {
+                    'valueUnsignedInt': 1,
+                    'url': 'https://fhir-ehr.cerner.com/r4/StructureDefinition/priority'
+                  }
+                ],
+                'url': 'https://fhir-ehr.cerner.com/r4/StructureDefinition/modifier-code'
+              },
+              {
+                'valueReference': {
+                  'reference': 'Location/683731',
```
Comment (alexbeckner1) [inline lib/resources/example_json/r4_examples_charge_item.rb:626]: Let's use location 31103071

File: lib/resources/example_json/r4_examples_charge_item.rb [hunk 12]
```diff
@@ -513,5 +513,221 @@ module Resources
...
+                  'display': '1 West SA2'
```
Comment (alexbeckner1) [inline lib/resources/example_json/r4_examples_charge_item.rb:627]: do we really support taking in a display here?

File: lib/resources/example_json/r4_examples_charge_item.rb [hunk 13]
```diff
@@ -513,5 +513,221 @@ module Resources
...
+              },
+              {
+                'extension': [
+                  {
+                    'id': '68985657',
+                    'valueString': 'Family Planning Indicator',
+                    'url': 'custom-attribute-name'
+                  },
+                  {
+                    'valueInteger': 689_856_57,
```
Comment (alexbeckner1) [inline lib/resources/example_json/r4_examples_charge_item.rb:639]: Is this a valid Interger value with the _ characters in it?

File: lib/resources/example_json/r4_examples_charge_item.rb [hunk 14]
```diff
@@ -513,5 +513,221 @@ module Resources
...
+                'valueReference': {
+                  'reference': 'Location/683731',
+                  'display': '1 West SA2'
+                },
+                'url': 'https://fhir-ehr.cerner.com/r4/StructureDefinition/performing-location'
+              },
+              {
+                'extension': [
+                  {
+                    'id': '68985657',
```
Comment (alexbeckner1) [inline lib/resources/example_json/r4_examples_charge_item.rb:634]: Do we actually support an id here for the custom-attribute-name on a create?

File: lib/resources/example_json/r4_examples_charge_item.rb [hunk 15]
```diff
@@ -513,5 +513,221 @@ module Resources
...
+                            {
+                              'valueCoding': {
+                                'system': 'https://fhir.cerner.com/ec2458f2-1e24-41c8-b71b-0e701af7583d/codeSet/14002',
+                                'code': '615216'
+                              },
+                              'url': 'https://fhir-ehr.cerner.com/r4/StructureDefinition/bill-code-schedule'
+                            }
+                          ],
+                          'system': 'https://fhir.cerner.com/ec2458f2-1e24-41c8-b71b-0e701af7583d/CodeSystem/BillCodes-PROCCODE',
+                          'code': '01U10JZ'
```
Comment (alexbeckner1) [inline lib/resources/example_json/r4_examples_charge_item.rb:690]: Let's use code GZ1 here

File: lib/resources/example_json/r4_examples_charge_item.rb [hunk 16]
```diff
@@ -513,5 +513,221 @@ module Resources
...
+                        {
+                          'valueCoding': {
+                            'system': 'https://fhir.cerner.com/ec2458f2-1e24-41c8-b71b-0e701af7583d/codeSet/14002',
+                            'code': '615217'
+                          },
+                          'url': 'https://fhir-ehr.cerner.com/r4/StructureDefinition/bill-code-schedule'
+                        }
+                      ],
+                      'system': 'https://fhir.cerner.com/ec2458f2-1e24-41c8-b71b-0e701af7583d/CodeSystem/BillCodes-REVENUE',
+                      'code': '0100'
```
Comment (alexbeckner1) [inline lib/resources/example_json/r4_examples_charge_item.rb:717]: let's use code 0900

**Review by Nandita92 — APPROVED** [review]

File: lib/resources/example_json/r4_examples_charge_item.rb (hunk 13 above)
Comment (krvicky31) [reply lib/resources/example_json/r4_examples_charge_item.rb:639]: I believe it's a valid integer as the underscore is provided to improve readability @SantanuBanerjee your opinions.

Comment (SantanuBanerjee) [reply lib/resources/example_json/r4_examples_charge_item.rb:639]: I think ruby lets us use the underscore (_) character to make large numbers more readable , mainly to separate thousands, but then we may want to decide if at all we want to use this ruby specific syntax in the json body.

File: lib/resources/r4/charge_item_create.yaml (hunk 1 above)
Comment (krvicky31) [reply lib/resources/r4/charge_item_create.yaml]: [changed here](https://github.com/cerner/fhir.cerner.com/pull/778/files#diff-2830ed2523973e82037a23daf274840e05a6c6e05b9f698fbd5f3e7fc24f57bbR36)

File: lib/resources/example_json/r4_examples_charge_item.rb (hunk 16 above)
Comment (krvicky31) [reply lib/resources/example_json/r4_examples_charge_item.rb]: [Updated](https://github.com/cerner/fhir.cerner.com/pull/778/files#diff-5cd46ad5f0af8a24217f67b656e77ce3624d6cefec3a18eddf7adc12bd47d35fR717)

File: lib/resources/example_json/r4_examples_charge_item.rb (hunk 15 above)
Comment (krvicky31) [reply lib/resources/example_json/r4_examples_charge_item.rb]: [updated](https://github.com/cerner/fhir.cerner.com/pull/778/files#diff-5cd46ad5f0af8a24217f67b656e77ce3624d6cefec3a18eddf7adc12bd47d35fR690)

File: lib/resources/example_json/r4_examples_charge_item.rb (hunk 11 above)
Comment (krvicky31) [reply lib/resources/example_json/r4_examples_charge_item.rb]: [Updated](https://github.com/cerner/fhir.cerner.com/pull/778/files#diff-5cd46ad5f0af8a24217f67b656e77ce3624d6cefec3a18eddf7adc12bd47d35fR626)

File: lib/resources/example_json/r4_examples_charge_item.rb (hunk 10 above)
Comment (krvicky31) [reply lib/resources/example_json/r4_examples_charge_item.rb]: [Updated](https://github.com/cerner/fhir.cerner.com/pull/778/files#diff-5cd46ad5f0af8a24217f67b656e77ce3624d6cefec3a18eddf7adc12bd47d35fR589)

File: lib/resources/example_json/r4_examples_charge_item.rb (hunk 9 above)
Comment (krvicky31) [reply lib/resources/example_json/r4_examples_charge_item.rb]: [updated](https://github.com/cerner/fhir.cerner.com/pull/778/files#diff-5cd46ad5f0af8a24217f67b656e77ce3624d6cefec3a18eddf7adc12bd47d35fR565)

File: lib/resources/example_json/r4_examples_charge_item.rb (hunk 7 above)
Comment (krvicky31) [reply lib/resources/example_json/r4_examples_charge_item.rb]: [updated](https://github.com/cerner/fhir.cerner.com/pull/778/files#diff-5cd46ad5f0af8a24217f67b656e77ce3624d6cefec3a18eddf7adc12bd47d35fR529)

File: lib/resources/example_json/r4_examples_charge_item.rb (hunk 6 above)
Comment (krvicky31) [reply lib/resources/example_json/r4_examples_charge_item.rb]: [updated](https://github.com/cerner/fhir.cerner.com/pull/778/files#diff-5cd46ad5f0af8a24217f67b656e77ce3624d6cefec3a18eddf7adc12bd47d35fR526)

File: lib/resources/r4/charge_item_create.yaml (hunk 5 above)
[the remark this answers is not in the document]
Comment (krvicky31) [reply lib/resources/r4/charge_item_create.yaml]: [actual notes updated here](https://github.com/cerner/fhir.cerner.com/pull/778/files#diff-2830ed2523973e82037a23daf274840e05a6c6e05b9f698fbd5f3e7fc24f57bbR179-R181)

File: lib/resources/example_json/r4_examples_charge_item.rb (hunk 13 above)
Comment (krvicky31) [reply lib/resources/example_json/r4_examples_charge_item.rb]: Replaced the value with less than thousand #diff-5cd46ad5f0af8a24217f67b656e77ce3624d6cefec3a18eddf7adc12bd47d35fR639

File: content/millennium/r4/financial/general/charge-item.md [hunk 17]
```diff
@@ -212,6 +212,61 @@ The `ETag` response header indicates the current `If-Match` version to use on su
...
+
+<%= beta_tag(action: true) %>
+
+Creates a charge event which can result in the creation of one or more ChargeItems.
+
+    POST /ChargeItem/:id/$create
+
+_Implementation Notes_
+
+* Only the body fields mentioned below are supported. Unsupported fields will be ignored.
```
Comment (alexbeckner1) [inline content/millennium/r4/financial/general/charge-item.md:225]: Let's remove the "unsupported fields will be ignored." as I don't think that is actually true.

File: lib/resources/r4/charge_item_create.yaml [hunk 18]
```diff
@@ -0,0 +1,426 @@
+---
```
Comment (alexbeckner1) [inline lib/resources/r4/charge_item_create.yaml:1]: Can you please review the structure and notes that have been done on the Modify feature fhir.cerner.com documentation to make sure the create is also in sync? https://github.com/cerner/fhir.cerner.com/pull/780/files?authenticity_token=uwEEH%2Bfxifl1rPvDH1HnciXQIox4yzgSotSahdkV2UUsj0ZFptY0ay1zE%2BRIEmy16hyXh%2BdzUH%2BM%2Be8kvpKvdg%3D%3D&file-filters%5B%5D=.rb&file-filters%5B%5D=.yaml#diff-3c565d9c41e2bb3d417efe63a774683e72b07cd38f2ff54a695c695999f29938

**Review by BhavyaAcharya — APPROVED** [review]

File: lib/resources/r4/charge_item_create.yaml (hunk 4 above)
Comment (alexbeckner1) [reply lib/resources/r4/charge_item_create.yaml]: Let's go ahead and remove date/dateTime for now. As we have an issue that doesn't allow us to support that at this time.

File: lib/resources/r4/charge_item_create.yaml (hunk 3 above)
Comment (alexbeckner1) [reply lib/resources/r4/charge_item_create.yaml]: Let's use the same notes from modify: https://github.com/cerner/fhir.cerner.com/pull/780/files?authenticity_token=C7RP2D9p%2BLf3pGVYHdXtojtlmvQ%2BP%2BZTtdxbBvqImhacOg2Cfk5FJa97jX9KlmZl9Kkv%2F6GHjj6b8S6nnQ%2FsJQ%3D%3D&file-filters%5B%5D=.rb&file-filters%5B%5D=.yaml#diff-3c565d9c41e2bb3d417efe63a774683e72b07cd38f2ff54a695c695999f29938

Comment (krvicky31) [reply lib/resources/r4/charge_item_create.yaml]: Updated here https://github.com/cerner/fhir.cerner.com/pull/778/files#diff-2830ed2523973e82037a23daf274840e05a6c6e05b9f698fbd5f3e7fc24f57bbR292-R294

File: lib/resources/r4/charge_item_create.yaml (hunk 5 above)
[the remark this answers is not in the document]
Comment (krvicky31) [reply lib/resources/r4/charge_item_create.yaml]: updated here https://github.com/cerner/fhir.cerner.com/pull/778/files#diff-2830ed2523973e82037a23daf274840e05a6c6e05b9f698fbd5f3e7fc24f57bbR292-R294

File: lib/resources/r4/charge_item_create.yaml (hunk 4 above)
Comment (krvicky31) [reply lib/resources/r4/charge_item_create.yaml]: updated here https://github.com/cerner/fhir.cerner.com/pull/778/files#diff-2830ed2523973e82037a23daf274840e05a6c6e05b9f698fbd5f3e7fc24f57bbR292-R294

File: lib/resources/r4/charge_item_create.yaml (hunk 18 above)
Comment (krvicky31) [reply lib/resources/r4/charge_item_create.yaml:1]: Both synced to similar https://github.com/cerner/fhir.cerner.com/pull/778/files#diff-2830ed2523973e82037a23daf274840e05a6c6e05b9f698fbd5f3e7fc24f57bbR2-R3

File: content/millennium/r4/financial/general/charge-item.md (hunk 17 above)
Comment (krvicky31) [reply content/millennium/r4/financial/general/charge-item.md]: Removed https://github.com/cerner/fhir.cerner.com/pull/778/files#diff-03f941e4ceb2839bfe0b49a5cc913e04919560e289603ae7b08b875f86ec568dR225

File: lib/resources/example_json/r4_examples_charge_item.rb (hunk 12 above)
Comment (krvicky31) [reply lib/resources/example_json/r4_examples_charge_item.rb:598]: @alexbeckner1 In modify also display is present https://github.com/cerner/fhir.cerner.com/pull/780/files#diff-5cd46ad5f0af8a24217f67b656e77ce3624d6cefec3a18eddf7adc12bd47d35fR608-R612 so do we need to remove from the create.

File: lib/resources/r4/charge_item_create.yaml [hunk 19]
```diff
@@ -0,0 +1,413 @@
...
+        }
+      }
+    note: >
+      <ul>
+        <li>Must be an Encounter reference</li>
+      </ul>
+    url: https://hl7.org/fhir/r4/chargeitem-definitions.html#ChargeItem.context
+
+  - name: occurrenceDateTime
+    required: 'No'
```
Comment (alexbeckner1) [inline lib/resources/r4/charge_item_create.yaml:117]: I think occurrenceDateTime is required for the create service.

File: lib/resources/r4/charge_item_create.yaml [hunk 20]
```diff
@@ -0,0 +1,413 @@
...
+    description: Date/time when the charged service was applied.
+    action: create
+    example: |
+      {
+        "occurrenceDateTime": "2021-08-02T21:53:34Z"
+      }
+    url: https://hl7.org/fhir/r4/chargeitem-definitions.html#ChargeItem.occurrence_x_
+
+  - name: quantity
+    required: 'No'
```
Comment (alexbeckner1) [inline lib/resources/r4/charge_item_create.yaml:128]: I think quantity is required for the create service

Comment (krvicky31) [reply lib/resources/r4/charge_item_create.yaml]: changed required to yes https://github.com/cerner/fhir.cerner.com/pull/778/commits/051542b5d68755b7c524626b97685248a6b576ae#diff-2830ed2523973e82037a23daf274840e05a6c6e05b9f698fbd5f3e7fc24f57bbR128

File: lib/resources/r4/charge_item_create.yaml (hunk 19 above)
Comment (krvicky31) [reply lib/resources/r4/charge_item_create.yaml]: changed required to yes https://github.com/cerner/fhir.cerner.com/pull/778/commits/051542b5d68755b7c524626b97685248a6b576ae#diff-2830ed2523973e82037a23daf274840e05a6c6e05b9f698fbd5f3e7fc24f57bbR117

**Comment by krvicky31** [comment]
@SantanuBanerjee Having some issue in login, he has reviewed the changes and approved.

**Review by NeethuKP — APPROVED** [review]

**Pull request merged**